ballista/rust/scheduler/src/state/task_manager.rs

```diff
@@ -196,21 +215,24 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> TaskManager<T, U>
             .collect();
 
         let mut assignments: Vec<(String, Task)> = vec![];
-        let mut pending_tasks = 0usize;
         let mut assign_tasks = 0usize;
+
         let job_cache = self.active_job_cache.read().await;
+
         for (_job_id, graph) in job_cache.iter() {
             let mut graph = graph.write().await;
             for reservation in free_reservations.iter().skip(assign_tasks) {
                 if let Some(task) = graph.pop_next_task(&reservation.executor_id)? {
                     assignments.push((reservation.executor_id.clone(), task));
                     assign_tasks += 1;
+                    self.decrease_pending_queue_size(1)?;
                 } else {
                     break;
                 }
             }
+
             if assign_tasks >= free_reservations.len() {
-                pending_tasks = graph.available_tasks();
+                self.increase_pending_queue_size(graph.available_tasks())?;
```

Comment:
I don't think you want this here. I think this would do some double counting. I think we only increment in `submit_job` and `update_job` (I think... the stage logic is pretty complicated to I need to look at it in more depth again)

Reply:
Good point